crates/worker/src/partition_processor_manager.rs

```diff
@@ -774,12 +774,13 @@ impl PartitionProcessorManager {
     fn on_control_processor(&mut self, control_processor: ControlProcessor) {
         let partition_id = control_processor.partition_id;
 
-        if control_processor.current_version
-            < self
-                .replica_set_states
-                .membership_state(partition_id)
-                .observed_current_membership
-                .version
+        if control_processor.current_version > Version::INVALID
```

Comment:
I think we actually want to ignore ControlProcessor messages coming from nodes not running the updated protocol. The impact is that we'll that disruption during upgrade/downgrade but it was deemed as an acceptable compromise when the original PR was authored.

Reply:
Ack, I was fixated on the downgrade path, but this was clearly a mistake! Fixed.